.github/workflows/release.yaml

```diff
@@ -3,46 +3,43 @@ on:
   push:
     tags:
       - 'v[0-9]+.[0-9]+.[0-9]+'
+  pull_request: {}
 defaults:
   run:
     shell: bash
 jobs:
-  create:
+  release:
     runs-on: ubuntu-latest
-    outputs:
-      upload_url: ${{ steps.release.outputs.upload_url }}
-    steps:
-      - uses: actions/create-release@v1
-        id: release
-        env:
-          GITHUB_TOKEN: ${{ github.token }}
-        with:
-          draft: true
-          tag_name: ${{ github.ref }}
-          release_name: ${{ github.ref }}
-  assets:
-    needs: create
-    strategy:
-      matrix:
-        os:
-          - ubuntu-latest
-          - macos-latest
-          - windows-latest
-    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
       - uses: actions/setup-go@v2
         with:
           go-version: '~1.16'
+
       - run: |
-          echo "asset_path=bin/opm" >> $GITHUB_ENV
-          echo "asset_name=$(go env GOOS)-$(go env GOARCH)-opm$(go env GOEXE)" >> $GITHUB_ENV
-      - run: make ${{ env.asset_path }}
-      - uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ github.token }}
+          git clone https://github.com/tpoechtrager/osxcross /tmp/osxcross
+          cd /tmp/osxcross
+          wget -P tarballs 'https://www.dropbox.com/s/cklcenl4667il6b/MacOSX11.1.sdk.tar.xz'
```

Comment:
I built and uploaded this file to dropbox. Downloading the SDK from Apple directly requires a logged in Apple developer account, which seems out of the question for a GH action.

Reply:
I'm just trying to wrap my head around this as I'm not a frequent mac user - is there any reason we couldn't just dedicate a job here to installing this macos sdk that `runs-on: macos-<some tag>`? I poked around the osxcross library real quick, but it looked like there's a script housed in that repository that should generate the sdk: https://github.com/tpoechtrager/osxcross/blob/master/tools/gen_sdk_package.sh. I need to dive deeper into that script and it might be beneficial to do a quick github search to see how others are handling the installment of this sdk.

Reply:
I think it would be possible to run osxcross's `gen_sdk_package.sh` on a macos github host and upload the resulting package via a `actions/upload-artifact@v2` step

Then in the linux job, we could download that artifact instead of doing the wget to dropbox. I _think_ caching would just work the same on the linux host, and we'd probably want to look into whether its possible to also cache the generated package on the macos host.

Reply:
Yeah I think that all sounds reasonable to me. I'm fine with keeping this implementation for now, but I'd like to move away from housing a custom dropbox file as using something more native is likely cleaner and easier to maintain going forward. I'm not super familiar with how caching works at a workflow vs. job level, but I'd assume that all jobs defined in an individual workflow share a local cache despite jobs using different machine os'?

Reply:
Sounds good. I vote to merge this PR without this change and do this work in a follow-up. I wouldn't be surprised to see some other minor issues pop up after merge since there's some logic in the workflow file that is hard to simulate locally.

Reply:
/hold while we find out if this is legal

Reply:
For now, I removed the darwin build targets, disabled GoReleaser's GitHub release steps, and re-introduces our existing github release workflow.

So the current state of this PR is just the addition of GoReleaser multi-arch image builds/pushes.